### DISCUSSION TITLE
Make DynamicMethod invokable "directly"

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/runtime#68135] [Ideas] [answered]
Hi all, code like the below currently fails:
```cs
var dm = new DynamicMethod(...); // emit some IL
var type = module.DefineType(...);
var mb = type.DefineMethod(...);
mb.GetILGenerator().Emit(OpCodes.Call, dm);

type.CreateType().GetMethod(...).Invoke(...); // exception: Unable to import a global method or field from a different module
```
Looking at the code, there is [GetMethodTokenNoLock](https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs) method and it doesn't support `DynamicMethod` at all. Is there some underlying limitation which prevents this from being a viable scenario? If not, can it be fixed, please? :)

Currently I end up having to create a delegate, store it in a static field somewhere, and then call the delegates `Invoke` method - which basically calls the `DynamicMethod` indirectly. Not only it's make work, but it's also a delegate invocation which is probably slightly slower than a direct call the `DynamicMethod`.

**Comment by teo-tsirpanis** [comment]
@dotnet/area-system-reflection-emit

**Reply by TahirAhmadov** [reply]
That link is broken?

**Reply by teo-tsirpanis** [reply]
It pings the area owners of Reflection.Emit, and can be used by .NET team members and community triagers (I am the latter). I will open an issue, hope it gets more attention.

**Comment by teo-tsirpanis** [comment, marked as answer]
Opened #68791.

**Reply by TahirAhmadov** [reply]
Thank you.
